src/opera/pge/disp_s1/disp_s1_pge.py

```diff
@@ -63,6 +64,45 @@ def run_preprocessor(self, **kwargs):
                                              self.runconfig.algorithm_parameters_schema_path,
                                              self.runconfig.algorithm_parameters_file_config_path,
                                              self.logger)
+        self.convert_troposphere_model_files()
+
+    def convert_troposphere_model_files(self):
+        """
+        Convert grip (.grp) files to netCDF (.nc)
+        Update the in-memory runconfig object such that the SAS/dynamic_ancillary_file_group/troposphere_files
+        section now points to the converted .nc files in the scratch directory.
+
+        """
+        # Retrieve the troposphere weather model file group from the run config file
+        troposphere_model_files_list = self.runconfig.sas_config['dynamic_ancillary_file_group']['troposphere_files']
+
+        # Converted files will be stored in the scratch directory.
+        scratch_dir = self.runconfig.sas_config['product_path_group']['scratch_path']
+
+        netcdf_file_list = []
+
+        for tropo_file in troposphere_model_files_list:
+            if tropo_file[-3:] == 'grb':
+                # change the extension to .nc
+                netcdf_file = join(scratch_dir, tropo_file.split('/')[-1][:-4] + '.nc')
```

Comment:
`splitext` could be similarly leveraged here: `netcdf_file = join(scratch_dir, os.path.splitext(tropo_file)[0] + '.nc')`